### ISSUE TITLE
Remove requirement to specify issue title in pull request

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#3540]
For some time now, GitHub is showing cards when we hover over an issue number which gives us a quick summary of the issue. The issue title is shown in the card among other things.

Given that and the fact that we already require the issue number to be specified in the pull request description, I believe it's now moot to ask the contributors to include the issue title in the pull request description. It's better if we remove this unnecessary burden from them.

What do you say?

**Comment by sivaraam** [comment]
FYI: A change for this is already present in the [`pull-request-template-change`](https://github.com/commons-app/apps-android-commons/tree/pull-request-template-change) branch of this repo.

**Comment by macgills** [comment]
Go for it

**Comment by 6point022** [comment]
While it is definitely unnecessary for desktop users, having the PR title in the description could be useful for mobile users, considering there's no quick summary option as of now. And the new GitHub apps could probably result in a lot more people using GitHub through mobile.

Also, while we are on subject of making changes to the PR template, can we also modify "**Screenshots showing what changed (optional - for UI changes)**." I am never sure of what it means. I don't know if I misunderstand it or it could be phrased better. Does it mean this is optional? Does it mean this is optional for UI changes? Does it mean this is optional, but necessary for UI changes?

**Comment by macgills** [comment]
Github is unusable on mobile for everything but commenting, I don't think we should make an affordance for a review context.

It means add screenshots if you changed the UI. We could change it to to `for UI changes only`. Separate issue though, please make an issue to discuss it further

**Comment by 6point022** [comment]
>Github is unusable on mobile for everything but commenting, I don't think we should make an affordance for a review context.

That is true, however, it may get better with the [official Android and iOS apps that have been released](https://www.theverge.com/2020/3/17/21184254/github-mobile-app-now-available-download-ios-android-microsoft).

> Separate issue though, please make an issue to discuss it further

Sure, will do that.

**Comment by sivaraam** [comment]
> Github is unusable on mobile for everything but commenting, I don't think we should make an affordance for a review context.

I agree with this. 

> That is true, however, it may get better with the official Android and iOS apps that have been released.

I've used the GitHub android app ever since the beta was launched. When it comes to reviewing PRs, trust me, it's still not there yet. Let's hope it gets better over time.

> Nevertheless, I think it's completely alright to remove the title requirement from the template.

👍 

> Also, while we are on the subject of making changes to the PR template, can we also modify "Screenshots showing what changed (optional - for UI changes)." I am never sure of what it means.

Don't worry. You're [not the only one who got confused](https://github.com/commons-app/apps-android-commons/pull/3496#issue-386861711). It really does need to be changed for the good. 🙂 We'll discuss that in #3544, though.

I believe we're find here with the change. I'll get the PR up, then.

**Comment by maskaravivek** [comment]
I am not sure if we should make this change. Personally I use the new Github mobile app a lot for keeping track of the active issues. 

Github mobile app recently came out of beta and they have been recently putting efforts to improve the mobile experience. 

IMO it doesn't hurt to keep both the title and the issue number. Even for desktops I would be lazy enough to go and hover over the issue number.

**Comment by sivaraam** [comment]
> IMO it doesn't hurt to keep both the title and the issue number. 

Personally, I've always found requesting the issue title to be a strange thing to request in the pull request description. Even if a person wants to know what is being fixed there's already a "What changes did you make and why?" section that should cover it. To add to that, a good PR would convey very well what is being fixed in the PR title and description itself. So, we really lose nothing by removing the issue title requirement, IMO. We just reduce what's possibly an unnecessary burden.

> Even for desktops I would be lazy enough to go and hover over the issue number.

Come on. People already say we're indolent just for sitting and working in front our desktops. Now think of what they would say if they know we feel lazy to even move the mouse! 😉  Ok, I was just kidding. But seriously, have you seen that hover card? If not, I really recommend you to give it a try. It's really beautiful and gives a nice summary of the linked issue. I love it. I believe you would too.

**Comment by macgills** [comment]
Personally when I start working on an issue I copy the title of an issue eg `Remove requirement to specify issue title in pull request #3540`, put the number at the start and then use that as the prefix for all my commits and if it is only 1 commit it ends up automatically being the title. Helps keep the context of my work so even if I have a stupid brain for a bit and just add the commit message `refactor` it gets seen as `#3540 Remove requirement to specify issue title in pull request - refactor`

I don't believe reviewing ever could or ever should be done on a phone, not until people start coding on phones anyhow but the raw required width makes that unattainable to me. Tablet is okay I guess.

**Comment by maskaravivek** [comment]
Okay lets remove the requirement. :)
